tests/backup/backup_resiliency_test.go

```diff
@@ -758,3 +759,166 @@ var _ = Describe("{CancelAllRunningBackupJobs}", func() {
+			}
+
+			// Incremental backup
+			for _, namespace := range bkpNamespaces {
+				incrementalBackupName = fmt.Sprintf("%s-%s-%v", "incremental-backup", namespace, time.Now().Unix())
+				incrementalBackupNames = append(incrementalBackupNames, incrementalBackupName)
+				err = CreateBackup(incrementalBackupName, SourceClusterName, customBackupLocationName, backupLocationUID, []string{namespace},
+					labelSelectors, orgID, clusterUid, "", "", "", "", ctx)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying backup [%s] creation", incrementalBackupName))
+			}
+			log.Infof("List of backups - %v", backupNames)
+			log.Infof("List of Incremental backups - %v", incrementalBackupNames)
+
+		})
+		Step("Deleting incremental backup", func() {
+			log.InfoD("Deleting incremental backups")
+			backupDriver := Inst().Backup
+			for _, backupName := range incrementalBackupNames {
+				ctx, err := backup.GetAdminCtxFromSecret()
+				log.FailOnError(err, "Fetching px-central-admin ctx")
+				log.Infof("About to delete backup - %s", backupName)
+				backupUID, err := backupDriver.GetBackupUID(ctx, backupName, orgID)
+				log.FailOnError(err, "Failed while trying to get backup UID for - %s", backupName)
+				_, err = DeleteBackup(backupName, backupUID, orgID, ctx)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Deleting backup - [%s]", backupName))
+			}
+		})
+		Step("Taking incremental backups of applications again", func() {
+			log.InfoD("Taking incremental backups of applications again")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			// Incremental backup
+			for _, namespace := range bkpNamespaces {
+				incrementalBackupName = fmt.Sprintf("%s-%s-%v", "incremental-backup", namespace, time.Now().Unix())
+				incrementalBackupNames = append(incrementalBackupNames, incrementalBackupName)
+				incrementalBackupNamesMap[incrementalBackupName] = incrementalBackupName
+				err = CreateBackup(incrementalBackupName, SourceClusterName, customBackupLocationName, backupLocationUID, []string{namespace},
+					labelSelectors, orgID, clusterUid, "", "", "", "", ctx)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying backup [%s] creation", incrementalBackupName))
+			}
+			log.Infof("List of New Incremental backups - %v", incrementalBackupNames)
+		})
+		Step("Check if backups are incremental backups or not", func() {
+			log.InfoD("Check if backups are incremental backups or not")
+			var bkp *api.BackupObject
+			backupDriver := Inst().Backup
+			bkpEnumerateReq := &api.BackupEnumerateRequest{
+				OrgId: orgID}
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx failed")
+			curBackups, err := backupDriver.EnumerateBackup(ctx, bkpEnumerateReq)
+			for _, bkp = range curBackups.GetBackups() {
+				for _, vol := range bkp.GetVolumes() {
+					if incrementalBackupNamesMap[bkp.GetName()] == bkp.GetName() {
```

Comment:
You may want to consider trying out @snigdha-px's IsPresent helper function for checking if something exists in a slice. Although maps can be useful for improving search efficiency.

Reply:
Map would be much faster to compute.

Reply:
Agreed.